dev/io.openliberty.microprofile.health.3.0.internal/src/io/openliberty/microprofile/health30/internal/HealthCheck30HttpResponseBuilder.java

```diff
@@ -94,19 +92,22 @@ protected void setChecks(HealthCheckResponse response) {
         check.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_NAME, response.getName());
 
         Status checkStatus = response.getStatus();
-        check.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_STATUS, checkStatus);
         if (checkStatus != null) {
             if (checkStatus.equals(Status.DOWN))
                 overallStatus = Status.DOWN;
         } else {
             if (tc.isDebugEnabled())
                 Tr.debug(tc, "setChecks(): checkStatus is null");
             overallStatus = Status.DOWN; // treat as fail case
+            checkStatus = Status.DOWN;
         }
 
+        check.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_STATUS, checkStatus.toString());
```

Comment:
Does this change behavior?  Before it would be null instead of DOWN.  Not sure why doing toString now when it wasn't done before.

Reply:
What I am suggesting here is:

```suggestion
        check.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_STATUS, checkStatus == null ? checkStatus : checkStatus.toString());
```

That is only if the toString() is required where it wasn't required before.

Reply:
If you need a String version of the enum, I think it is more appropriate to use `name()` instead of `toString()`

Reply:
If it is not a String, I get:
```
java.lang.Throwable: java.lang.IllegalArgumentException: Type class org.eclipse.microprofile.health.HealthCheckResponse$Status is not supported.
```

I have changed to .name() and kept the behavior the same as before.